extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ImageReference.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.quarkus.container.spi;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Strings;
+
+/**
+ * This is basically a simplified version of {@code com.google.cloud.tools.jib.api.ImageReference}
+ */
+public class ImageReference {
```

Comment:
+1 we might want to replace our ImageUtils with that.

Reply:
Good idea! 

Let's do that in a follow up. Perhaps the PR you are working on that reworks S2I?